src/pages/RFOX/components/Bridge/BridgeConfirm.tsx

```diff
@@ -0,0 +1,416 @@
+  )
+
+  const bridgeAmountCryptoPrecision = useMemo(
+    () => fromBaseUnit(bridgeQuote.bridgeAmountCryptoBaseUnit, sellAsset?.precision ?? 0),
+    [bridgeQuote.bridgeAmountCryptoBaseUnit, sellAsset?.precision],
+  )
+
+  const sellAssetMarketDataUserCurrency = useAppSelector(state =>
+    selectMarketDataByAssetIdUserCurrency(state, bridgeQuote.sellAssetId),
+  )
+
+  const bridgeAmountUserCurrency = useMemo(
+    () =>
+      bnOrZero(bridgeAmountCryptoPrecision).times(sellAssetMarketDataUserCurrency.price).toFixed(),
+    [bridgeAmountCryptoPrecision, sellAssetMarketDataUserCurrency.price],
+  )
+
+  const accountNumberFilter = useMemo(
+    () => ({ assetId: bridgeQuote.sellAssetId, accountId: bridgeQuote.sellAssetAccountId }),
+    [bridgeQuote.sellAssetAccountId, bridgeQuote.sellAssetId],
+  )
+  const accountNumber = useAppSelector(state =>
+    selectAccountNumberByAccountId(state, accountNumberFilter),
+  )
+
+  const { data: quote, isLoading: isBridgeQuoteLoading } = useQuery({
+    ...reactQueries.swapper.arbitrumBridgeTradeQuote({
+      sellAsset: sellAsset!,
+      buyAsset: buyAsset!,
+      chainId: sellAsset!.chainId as EvmChainId,
+      sellAmountIncludingProtocolFeesCryptoBaseUnit: bridgeQuote.bridgeAmountCryptoBaseUnit,
+      affiliateBps: '0',
+      potentialAffiliateBps: '0',
+      allowMultiHop: true,
+      receiveAddress: fromAccountId(bridgeQuote.buyAssetAccountId).account,
+      sendAddress: fromAccountId(bridgeQuote.sellAssetAccountId).account,
+      accountNumber: accountNumber!,
+      wallet: wallet!,
+    }),
+    enabled: Boolean(sellAsset && buyAsset && accountNumber !== undefined && wallet),
+  })
+
+  const allowanceContract = useMemo(() => {
+    if (!quote || quote.isErr()) return
+
+    const tradeQuote = quote.unwrap()
+
+    return tradeQuote.steps[0].allowanceContract
+  }, [quote])
+
+  const { data: allowanceData, isLoading: isAllowanceDataLoading } = useAllowance({
+    assetId: bridgeQuote.sellAssetId,
+    spender: allowanceContract,
+    from: fromAccountId(bridgeQuote.sellAssetAccountId).account,
+  })
+
+  const isApprovalRequired = useMemo(
+    () => bnOrZero(allowanceData).lt(bridgeQuote.bridgeAmountCryptoBaseUnit),
+    [allowanceData, bridgeQuote.bridgeAmountCryptoBaseUnit],
+  )
+
+  const approvalCallData = useMemo(() => {
+    if (!allowanceContract) return
+
+    return encodeFunctionData({
+      abi: erc20ABI,
+      functionName: 'approve',
+      args: [getAddress(allowanceContract), BigInt(bridgeQuote.bridgeAmountCryptoBaseUnit)],
+    })
+  }, [allowanceContract, bridgeQuote.bridgeAmountCryptoBaseUnit])
+
+  const {
+    mutate: sendApprovalTx,
+    isPending: isApprovalMutationPending,
+    isSuccess: isApprovalMutationSuccess,
+    isIdle: isApprovalMutationIdle,
+  } = useMutation({
+    ...reactQueries.mutations.approve({
+      assetId: bridgeQuote.sellAssetId,
+      spender: allowanceContract!,
```

Comment:
needs handling without non-null assert

Reply:
Tackled in p3 8c3e2386ed

Note this has been handled in `handleApprove` (pun intended). The reason is a mutation isn't a query, and it runs JIT, user-initiated. So there is no `enabled` for it, however, it's the responsibility of the consumer to ensure all dependencies are met to run it.

Reply:
Now clean and type-safety 🎉  https://github.com/shapeshift/web/pull/7020#discussion_r1631768133